client/test/unit/specs/utils/projectStaffingNeededDate.spec.js

```diff
@@ -56,27 +58,71 @@ describe('Unit | Utils | Project Staffing Needed Date', () => {
       const expectedSortedJobs = [todayJob1, todayJob2, todayJob3, todayJob4, todayJob5];
 
       // When
-      const sortedJobs = projectStaffingNeededDate.sort(givenJobs);
+      const sortedJobs = projectStaffingNeededDate.sortAll(givenJobs);
       // Then
       expect(sortedJobs).to.deep.equal(expectedSortedJobs);
     });
+  });
+
+  describe('#sortAfter', () => {
+    let clock;
+    const availabityDate = moment(new Date(2017, 9, 3));
+
+    beforeEach(() => {
+      clock = sinon.useFakeTimers(new Date(2017, 9, 4).getTime());
+    });
+
+    afterEach(() => {
+      clock.restore();
+    });
 
-    it('should sort jobs according to project staffing needed date', () => {
+    it('should sort chronologically two jobs according to project staffing needed date', () => {
       // Given
       const futureJob = { activity: { id: 1, staffing_needed_from: '2017-11-03' } };
-      const beforeYesterdayJob = { activity: { id: 2, staffing_needed_from: '2017-10-02' } };
-      const todayJob = { activity: { id: 3, staffing_needed_from: '2017-10-04' } };
-      const veryOldJob = { activity: { id: 4, staffing_needed_from: '2017-09-15' } };
-      const yesterdayJob = { activity: { id: 5, staffing_needed_from: '2017-10-03' } };
-      const oldJob = { activity: { id: 6, staffing_needed_from: '2017-09-08' } };
+      const todayJob = { activity: { id: 2, staffing_needed_from: '2017-10-04' } };
+
+      const givenJobs = [futureJob, todayJob];
+      const expectedSortedJobs = [todayJob, futureJob];
+
+      // When
+      const sortedJobs = projectStaffingNeededDate.sortAfter(givenJobs, availabityDate);
 
+      // Then
+      expect(sortedJobs).to.deep.equal(expectedSortedJobs);
+    });
+
+    it('should sort chronologically five jobs according to project staffing needed date', () => {
```

Comment:
C'est 6 jobs, pas 5.

Expliciter dans le titre le fait qu'un doit être exclu et pourquoi ?

Reply:
Merci du retour

Reply:
should sort three jobs whose staffing_needed_from chronologically after availability date